### PR TITLE
Adding tmt packit automation 

### PR DESCRIPTION
**Pull request by Yarboa**
Tests are run by packit through tmt in testsing-farm Adding integration plan and tier-0 test verifying
initoverlayfs setup

### CODE REVIEW

**Comment by ericcurtin** [comment]
This is really neat @Yarboa

**Review by ericcurtin — APPROVED** [review]

**Comment by dougsland** [comment]
LGTM

**Comment by ericcurtin** [comment]
This should work after we merge this I think:

https://github.com/containers/initoverlayfs/pull/30

we need PR number 30 anyway for unaccounted devices.

File: .packit.yaml
Comment (lsm5) [inline .packit.yaml:6]: remove line 6 entirely. Given the version tagged doesn't begin with `v`, packit can just use the defaults.

That's currently causing the build failure as seen at https://copr.fedorainfracloud.org/coprs/packit/containers-initoverlayfs-28/build/6721833/

**Review by lsm5 — CHANGES_REQUESTED** [review]

**Review by mwperina — APPROVED** [review]
LGTM

**Comment by ericcurtin** [comment]
On this @mwperina 

"ccount mwperina has no write access nor is author of PR!"

I added you as a member with write access, I think you just have to accept.

**Comment by mwperina** [comment]
> On this @mwperina
> 
> "ccount mwperina has no write access nor is author of PR!"
> 
> I added you as a member with write access, I think you just have to accept.

I probably missed some email, but I was able to find out the invite in my organizations and it works fine.
Thanks Eric.

**Comment by Yarboa** [comment]
There is an issue with reboot
```
            00:00:25 /integration/tests/boot-validation [1/1]
            00:00:25 errr /integration/tests/boot-validation (on default-0) (reboot timeout) [1/1]
        fail: Failed to pull workdir from the guest. This usually means that login as 'root' to the guest does not work
 ``` 
 I will have to reserve AWS and debug

**Comment by Yarboa** [comment]
Ok i was running it manually, 
https://artifacts.dev.testing-farm.io/fc6e7439-35be-41b0-a5af-a40f4c2ca703/
It seems to be TF issue, lets wait to this PR to pass all tests

**Comment by ericcurtin** [comment]
If we rebase this it might just work @Yarboa

**Comment by mwperina** [comment]
I just noticed that RPMs are generated as they would be a stable release (`initoverlayfs-0.98-1.el9.src.rpm`) and not as a snapshot build (`initoverlayfs-0.98-0.202312131810.gitefcc221.el9.src.rpm`). That should be fixed, if we want to include those RPMs into some snaphot builds repo (for example under centos-automotive-sig group) and make them consumable for other testers

**Comment by Yarboa** [comment]
> I just noticed that RPMs are generated as they would be a stable release (`initoverlayfs-0.98-1.el9.src.rpm`) and not as a snapshot build (`initoverlayfs-0.98-0.202312131810.gitefcc221.el9.src.rpm`). That should be fixed, if we want to include those RPMs into some snaphot builds repo (for example under centos-automotive-sig group) and make them consumable for other testers

@mwperina thanks
do you refer running to something like this ? build-scripts/version.sh

**Comment by mwperina** [comment]
> > I just noticed that RPMs are generated as they would be a stable release (`initoverlayfs-0.98-1.el9.src.rpm`) and not as a snapshot build (`initoverlayfs-0.98-0.202312131810.gitefcc221.el9.src.rpm`). That should be fixed, if we want to include those RPMs into some snaphot builds repo (for example under centos-automotive-sig group) and make them consumable for other testers
> 
> @mwperina thanks do you refer running to something like this ? build-scripts/version.sh

So BlueChi has that support, so if you want to inspire, please take a look at those files:

https://github.com/eclipse-bluechi/bluechi/blob/main/build-scripts/build-rpm.sh
https://github.com/eclipse-bluechi/bluechi/blob/main/build-scripts/build-srpm.sh
https://github.com/eclipse-bluechi/bluechi/blob/main/build-scripts/create-archive.sh
https://github.com/eclipse-bluechi/bluechi/blob/main/build-scripts/create-spec.sh
https://github.com/eclipse-bluechi/bluechi/blob/main/build-scripts/version.sh

https://github.com/eclipse-bluechi/bluechi/blob/main/.packit.yml#L20

But maybe it can be simplifed for initoverlayfs, which is much smaller project

**Comment by Yarboa** [comment]
> > > I just noticed that RPMs are generated as they would be a stable release (`initoverlayfs-0.98-1.el9.src.rpm`) and not as a snapshot build (`initoverlayfs-0.98-0.202312131810.gitefcc221.el9.src.rpm`). That should be fixed, if we want to include those RPMs into some snaphot builds repo (for example under centos-automotive-sig group) and make them consumable for other testers
> > 
> > 
> > @mwperina thanks do you refer running to something like this ? build-scripts/version.sh
> 
> So BlueChi has that support, so if you want to inspire, please take a look at those files:
> 
> https://github.com/eclipse-bluechi/bluechi/blob/main/build-scripts/build-rpm.sh https://github.com/eclipse-bluechi/bluechi/blob/main/build-scripts/build-srpm.sh
> 
> Above 2 files are not needed for packit, they are here for previous GH build/test integration and for local command line convenience.
> 
> Below files are required for packit integration: https://github.com/eclipse-bluechi/bluechi/blob/main/build-scripts/create-archive.sh https://github.com/eclipse-bluechi/bluechi/blob/main/build-scripts/create-spec.sh https://github.com/eclipse-bluechi/bluechi/blob/main/build-scripts/version.sh https://github.com/eclipse-bluechi/bluechi/blob/main/.packit.yml#L20
> 
> But maybe it can be simplifed for initoverlayfs, which is much smaller project

Thanks, much appreciated

**Comment by mwperina** [comment]
Unfortunately it's not enough, you still need to convert spec file to a template and replace version/release in this template per build:
https://github.com/eclipse-bluechi/bluechi/blob/main/build-scripts/create-spec.sh

**Review by ericcurtin — APPROVED** [review]
LGTM, we can do follow on work in the next PR if this is green

**Comment by ericcurtin** [comment]
I wish there was a auto-merge if build is green button here like we have in gitlab, it's useful...

**Comment by ericcurtin** [comment]
I found it in settings and turned it on, maybe the next PR will have the option

**Comment by Yarboa** [comment]
> Unfortunately it's not enough, you still need to convert spec file to a template and replace version/release in this template per build: https://github.com/eclipse-bluechi/bluechi/blob/main/build-scripts/create-spec.sh

Noticed that. yes, lets wait

**Comment by Yarboa** [comment]
verified 'package_version': '0.98-0.202312141231.git4c998f9',

**Pull request merged**
